### PR TITLE
feat: allow audiences for justification

### PR DESCRIPTION
**Pull request by yolocs**
Fix: https://github.com/abcxyz/jvs/issues/22

### CODE REVIEW

File: pkg/justification/processor.go [hunk 1]
```diff
@@ -63,7 +63,8 @@ func NewProcessor(kms *kms.KeyManagementClient, config *config.JustificationConf
 }
 
 const (
-	cacheKey = "signer"
+	cacheKey        = "signer"
+	DefaultAudience = "jvs"
```
Comment (sethvargo) [inline pkg/justification/processor.go:67]: These are usually URLs or reverse DNS like `com.abcxyz.jvs`

**Review by myurtoglu — APPROVED** [review]

File: pkg/justification/processor.go [hunk 2]
```diff
@@ -163,8 +164,14 @@ func (p *Processor) createToken(ctx context.Context, now time.Time, req *jvspb.C
 	exp := now.Add(req.Ttl.AsDuration())
 	justs := req.Justifications
 
+	// Use audiences in the request if provided.
+	aud := []string{DefaultAudience}
```
Comment (sethvargo) [inline pkg/justification/processor.go:168]: I might just be me, but this feels weird to alloc the aud that you might not use. Consider:

```go
aud := req.Audiences
if len(aud) == 0 {
  aud = []string{DefaultAudience}
}
```

File: protos/v0/jvs_request.proto [hunk 3]
```diff
@@ -25,6 +25,9 @@ option go_package = "github.com/abcxyz/jvs/apis/v0";
 message CreateJustificationRequest {
   repeated Justification justifications = 1;
   google.protobuf.Duration ttl = 2;
+
+  // Optional audiences for the justifications.
+  repeated string audiences = 3;
```
Comment (sethvargo) [inline protos/v0/jvs_request.proto:30]: Do you need to regen the protos?

Comment (yolocs) [reply protos/v0/jvs_request.proto:30]: Yes, did that in the PR.

File: pkg/justification/processor.go [hunk 4]
```diff
@@ -63,7 +63,8 @@ func NewProcessor(kms *kms.KeyManagementClient, config *config.JustificationConf
 }
 
 const (
-	cacheKey = "signer"
+	cacheKey        = "signer"
+	DefaultAudience = "dev.abcxyz.jvs"
```
Comment (sethvargo) [inline pkg/justification/processor.go:67]: Nit: document

**Review by sethvargo — APPROVED** [review]

**Pull request merged**